### ISSUE TITLE
"path not taken: false" even when there is no else clause...

### ISSUE THREAD

**Issue by jcward** [nadako/coverme#4]
The following if block gets a "path not taken: false" error. However, there is no else block... So, does it matter? Technically, there is no code here that was not covered:

![image](https://user-images.githubusercontent.com/2192439/43100529-d5e25290-8e82-11e8-950a-56232ca6f325.png)

...though I suppose if there were other expressions in the if statement, those would have been uncovered. For example: `if ( never_false && uh_oh_never_called() ) { }` -- But then again, those expressions themselves would have got their own counters, right?

Hmm...

**Comment by nadako** [comment]
> Technically, there is no code here that was not covered:

yes, but it's not about just covering lines of code, but covering all possible behaviours. we should always cover all conditions, otherwise there'll be a situation where a condition doesn't hold, so the code is not executed which will break further code :) also this helps finding redundant conditions sometimes!
